substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/agent/Agent2.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, 2024, Alibaba Group Holding Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.test.agent;
+
+import org.graalvm.nativeimage.ImageInfo;
+
+import java.lang.instrument.Instrumentation;
+
+public class Agent2 {
+    @SuppressWarnings("unused")
+    public static void premain(
+                    String agentArgs, Instrumentation inst) {
+        if (agentArgs != null && !agentArgs.isBlank()) {
+            String[] argPairs = agentArgs.split(",");
+            for (String argPair : argPairs) {
+                String[] pair = argPair.split("=");
+                if (pair.length == 2) {
+                    System.setProperty(pair[0], pair[1]);
+                }
+            }
+        }
+        System.setProperty("instrument.enable", "true");
```

Comment:
I would add a set of assertions that checks that the values for `Instrumentation` are correct at image run time. This way we assure nobody changes those values and surprises the users.

Reply:
The check of Instrumentation class is put in the Agent.

Reply:
I would still add a check for all the relevant methods that can be invoked on the `Instrumentation` good examples are: `Instrumentation#isModifiableClass`, `Instrumentation#isRetransformClassesSupported`, `Instrumentation#isModifiableModule`, `Instrumentation#getAllLoadedClasses`, etc.